packages/frontend/src/components/FolderList/Folders.tsx

```diff
@@ -0,0 +1,170 @@
+import { Box, Button, TextField, Typography } from '@mui/material';
+import axios from 'axios';
+import { t } from 'i18next';
+import { useEffect, useState } from 'react';
+import { useLocation } from 'react-router-dom';
+import { getCategoryRouteName } from '../../routes';
+import { ButtonWrapper } from '../../styles/common';
+import { theme } from '../../styles/createTheme';
+import { Modal } from '../Modal/Modal';
+import { FolderList } from './FolderList';
+
+interface Component {
+  categoryComponentId: string;
+  title: string;
+  alfrescoNodeId: string;
+}
+
+interface FoldersProps {
+  isEditing: boolean;
+}
+
+export const Folders = ({ isEditing }: FoldersProps) => {
+  const [components, setComponents] = useState([]);
+  const [open, setOpen] = useState(false);
+  const [success, setSuccess] = useState(false);
+  const [error, setError] = useState(false);
+  const [title, setTitle] = useState('');
+  const [selectedComponent, setSelectedComponent] = useState<Component | null>(null);
+  const [isNewList, setIsNewList] = useState(false);
+  const categoryName = getCategoryRouteName(useLocation());
+
+  const getComponents = async () => {
+    try {
+      const response: any = await axios.get(`/api/database/components/${categoryName}`);
+      setComponents(response.data.map((component: any) => component.node));
+    } catch (error) {
+      console.log(error);
+    }
+  };
+
+  useEffect(() => {
+    getComponents();
+  }, []);
+
+  const resetState = () => {};
+
+  const addList = async (title: string) => {
+    try {
+      const response = await axios.post(`/api/alfresco/folder/${categoryName}`, {
```

Comment:
General idea, not just here: Could this beginning bit be in a consts-type of file? It repeats quite often in slight variations. Not something that needs to be fixed here, but something to think about.